integration-tests/src/test/java/tachyon/LocalTachyonClusterResource.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon;
+
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import tachyon.conf.TachyonConf;
+import tachyon.master.LocalTachyonCluster;
+
+/**
+ * A JUnit Rule resource for automatically managing a local tachyon cluster for testing
+ */
+public class LocalTachyonClusterResource implements TestRule {
+
+  /** The capacity of the worker in bytes */
+  private final long mWorkerCapacityBytes;
+  /** The quota of space each user can request */
+  private final int mQuotaUnitBytes;
+  /** Block size for a user */
+  private final int mUserBlockSize;
+  /**
+   * If true (default), we start the cluster before running a test method. Otherwise, the method
+   * must start the cluster explicitly.
+   */
+  private final boolean mStartCluster;
+  /** Configuration parameters for the TachyonConf object used in the cluster */
+  private final String[] mConfParams;
+
+  /** The tachyon cluster being managed */
+  private LocalTachyonCluster mLocalTachyonCluster = null;
+  /** The configuration object used by the cluster */
+  private TachyonConf mTestConf = null;
+
+  /**
+   * Create a new instance.
+   *
+   * @param workerCapacityBytes the capacity of the worker in bytes
+   * @param quotaUnitBytes the quota of space each user can request
+   * @param userBlockSize the block size for a user
+   * @param startCluster whether or not to start the cluster before the test method starts
+   * @param confParams specific tachyon configuration parameters, specified as a list of strings,
+   *        treated as key-value pairs
+   */
+  public LocalTachyonClusterResource(long workerCapacityBytes, int quotaUnitBytes,
+      int userBlockSize, boolean startCluster, String... confParams) {
```

Comment:
Not sure what you mean. I thought it would be nice to have the same syntax for configuration in the constructor as it is in the annotation types, and you can't really pass a map into the annotation. For annotations, the only complex types you can have are other annotation types, so if we wanted some key value mapping, it would be something like
```
public @interface TachyonConfPair {
    String key();
    String value();
}
```

I tried this, and it ended up getting pretty verbose, so it's probably not worth the trouble.

Or are you thinking of something else?